master/internal/api_user.go

```diff
@@ -683,3 +683,116 @@ func (a *apiServer) PostUserActivity(
 	}
 	return &apiv1.PostUserActivityResponse{}, err
 }
+
+func (a *apiServer) PostLongLivedToken(
+	ctx context.Context, req *apiv1.PostLongLivedTokenRequest,
+) (*apiv1.PostLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanCreateUsersOwnToken(
+		ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenExpiration := time.Now().Add(user.TokenExpirationDuration)
+	if req.Lifespan != "" {
+		d, err := time.ParseDuration(req.Lifespan)
+		if err != nil || d < 0 {
+			return nil, status.Error(codes.InvalidArgument,
+				"Lifespan must be a Go-formatted duration string with a positive value")
+		}
+		tokenExpiration = time.Now().Add(d)
```

Comment:
Same as above `time.Now()` reports the current time in local time, so it could be a solid move to swap add a `time.Now().UTC()` to make sure that we keep things in a standard timezone

Reply:
Done